tests/gateway/test_client.py

```diff
@@ -471,3 +471,26 @@ def test_cllient_query_with_disallowed_param(mixed_gateway):
 
     with pytest.raises(HTTPError, match=".*The parameter 'model' is not permitted.*"):
         gateway_client.query(route=route.name, data=data)
+
+
+def test_query_timeout_not_retried(mixed_gateway):
+    gateway_client = MlflowGatewayClient(gateway_uri=mixed_gateway.url)
+
+    data = {"prompt": "Test", "temperature": 0.4}
+    route = "completions"
+
+    with mock.patch(
+        "mlflow.gateway.constants.MLFLOW_GATEWAY_CLIENT_QUERY_TIMEOUT_SECONDS",
+        new_callable=lambda: 1,
```

Comment:
Why do we need `new_callable` here?

Reply:
we don't. we just need `new=1`. Good catch.